install-devstack-xen.sh

```diff
@@ -4,11 +4,12 @@ set -eu
 function print_usage_and_die
 {
 cat >&2 << EOF
-usage: $0 XENSERVER XENSERVER_PASS PRIVKEY <optional arguments>
+usage: $0 XENSERVER XENSERVER_PASS PRIVKEY NODE_TYPE NODE_NAME [CONTROLLER_IP] <optional arguments>
```

Comment:
one question: So the controller typed node will always has all services enabled. right? How about we use two different templates for controller and compute nodes? So just a simple replacement basing on the type?